src/analyze/BuildSystem.re

```diff
@@ -120,6 +124,17 @@ let detect = (rootPath, bsconfig) => {
   }) : Bsb(bsbVersion);
 };
 
+let getEsyCompiledBase = esyVersion =>
```

Comment:
Thanks, I was looking to make this change but now I don't have to!

I think you could just examine the directoreis to see which are available and give precedence to `_esy` over `_build`. That way there's less coupling on the esy binary being available.
The only requirement would be a directory structure be in place.

Reply:
Why do you think it's better than reading `$cur__target_dir` from the `esy command-env --json` output? I think this is the most robust way — it was available from the beginning — 0.0.x, 0.1.x, 0.2.x, 0.3.x had it unchanged.